src/main/java/org/myrobotlab/programab/XmlParser.java

```diff
@@ -0,0 +1,31 @@
+package org.myrobotlab.programab;
+
+import org.myrobotlab.programab.models.Sraix;
+import org.myrobotlab.programab.models.Template;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.dataformat.xml.XmlMapper;
+
+/**
+ * Thread safe fasterjackson xml parser.
+ * 
+ * @author GroG
+ *
+ */
+public class XmlParser {
+
+  public static Template parseTemplate(String xml) throws JsonMappingException, JsonProcessingException {
+    ThreadLocal<XmlMapper> xmlMapperThreadLocal = ThreadLocal.withInitial(XmlMapper::new);
```

Comment:
it was suggested as a way to make the xml parser threadsafe, making it static would defeat that no ?

Reply:
The thread local needs to be static, then when you call the `get()` method in the code it returns or creates a new one that is only usable for that thread. The way it is now, the thread local wrapper is created every time and then garbage collected at the end of the method, so it can't reuse a previously created parser in the same thread if the method is called multiple times because the wrapper that keeps track of it is gone.

Technically, it's thread safe still, but only because it recreates the parser on every invocation of the method, which you could do without the thread local wrapper

Reply:
Thanks ! :+1: